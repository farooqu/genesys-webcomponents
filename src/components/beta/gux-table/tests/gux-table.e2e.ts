import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { a11yCheck } from '../../../../../tests/e2eTestUtils';

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();
  await page.addScriptTag({
    path: 'node_modules/axe-core/axe.min.js'
  });
  await page.waitForChanges();

  return page;
}

describe('gux-table-beta', () => {
  const tableContent = `
    <table slot="data">
      <thead>
        <tr>
          <th>First name</th>
          <th>Last name</th>
          <th data-cell-numeric>Age</th>
          <th data-cell-action>Action</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>John</td>
          <td>Doe</td>
          <td data-cell-numeric>25</td>
          <td data-cell-action>Delete</td>
        </tr>
        <tr>
          <td>Jane</td>
          <td>Doe</td>
          <td data-cell-numeric>23</td>
          <td data-cell-action>Delete</td>
        </tr>
        <tr>
          <td>Jane</td>
          <td>Doe</td>
          <td data-cell-numeric>21</td>
          <td data-cell-action>Delete</td>
        </tr>
        <tr>
          <td>Jane</td>
          <td>Doe</td>
          <td data-cell-numeric>23</td>
          <td data-cell-action>Delete</td>
        </tr>
      </tbody>
    </table>
  `;

  describe('#render', () => {
    [
      {
        description: 'empty table',
        html: `
        <gux-table-beta lang="en">
          <table slot="data">
            <thead>
              <tr>
                <th>First name</th>
                <th>Last name</th>
                <th data-cell-numeric>Age</th>
                <th data-cell-action>Action</th>
              </tr>
            </thead>
          </table>
        </gux-table-beta>
        `
      },
      {
        description: 'should render data table',
        html: `<gux-table-beta lang="en">${tableContent}</gux-table-beta>`
      },
      {
        description: 'should render compact data table',
        html: `<gux-table-beta compact lang="en">${tableContent}</gux-table-beta>`
      },
      {
        description: 'should render object table',
        html: `<gux-table-beta object-table lang="en">${tableContent}</gux-table-beta>`
      },

      {
        description: 'should render table with vertical scroll',
        html: `<gux-table-beta style="height: 150px" lang="en">${tableContent}</gux-table-beta>`
      },
      {
        description: 'should render table with horizontal scroll',
        html: `<gux-table-beta style="width: 300px" lang="en">${tableContent}</gux-table-beta>`
      },
      {
        description: 'should render table with rows selection',
        html: `<gux-table-beta object-table selectable-rows lang="en">${tableContent}</gux-table-beta>`
      },
      {
        description: 'should render empty table with rows selection',
        html: `
          <gux-table-beta object-table selectable-rows lang="en">
            <table slot="data">
              <thead>
                <tr>
                  <th>First name</th>
                  <th>Last name</th>
                  <th data-cell-numeric>Age</th>
                  <th data-cell-action>Action</th>
                </tr>
              </thead>
            </table>
          </gux-table-beta>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newNonrandomE2EPage({ html });
        const element = await page.find('gux-table-beta');
        await a11yCheck(page);

        expect(element).toHaveClass('hydrated');
        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
