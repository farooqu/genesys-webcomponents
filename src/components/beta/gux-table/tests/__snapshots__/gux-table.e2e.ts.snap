// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-table-beta #render empty table 1`] = `"<gux-table-beta lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> </table></div><div class=\\"gux-empty-table\\"><h2>No data available</h2></div></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render compact data table 1`] = `"<gux-table-beta compact=\\"\\" lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container gux-compact\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> <tbody> <tr> <td>John</td> <td>Doe</td> <td data-cell-numeric>25</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>21</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> </tbody> </table></div></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render data table 1`] = `"<gux-table-beta lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> <tbody> <tr> <td>John</td> <td>Doe</td> <td data-cell-numeric>25</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>21</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> </tbody> </table></div></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render empty table with rows selection 1`] = `"<gux-table-beta object-table=\\"\\" selectable-rows=\\"\\" lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container gux-object-table\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> </table></div><div class=\\"gux-empty-table\\"><h2>No data available</h2></div></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render object table 1`] = `"<gux-table-beta object-table=\\"\\" lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container gux-object-table\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> <tbody> <tr> <td>John</td> <td>Doe</td> <td data-cell-numeric>25</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>21</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> </tbody> </table></div></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render table with horizontal scroll 1`] = `"<gux-table-beta lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table gux-horizontal-scroll\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> <tbody> <tr> <td>John</td> <td>Doe</td> <td data-cell-numeric>25</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>21</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> </tbody> </table></div><gux-button-slot-beta accent=\\"secondary\\" class=\\"hydrated\\"><button class=\\"gux-table-scroll-left\\" disabled=\\"\\"><gux-icon icon-name=\\"chevron-left\\" screenreader-text=\\"Scroll Left\\" class=\\"hydrated\\"></gux-icon></button></gux-button-slot-beta><gux-button-slot-beta accent=\\"secondary\\" class=\\"hydrated\\" style=\\"margin-right: 0px;\\"><button class=\\"gux-table-scroll-right\\" style=\\"margin-right: 0px;\\"><gux-icon icon-name=\\"chevron-right\\" screenreader-text=\\"Scroll Right\\" class=\\"hydrated\\"></gux-icon></button></gux-button-slot-beta></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render table with rows selection 1`] = `"<gux-table-beta object-table=\\"\\" selectable-rows=\\"\\" lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table\\"><div tabindex=\\"-1\\" id=\\"gux-table-i\\" class=\\"gux-table-container gux-object-table\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> <tbody> <tr> <td>John</td> <td>Doe</td> <td data-cell-numeric>25</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>21</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> </tbody> </table></div></div></gux-table-beta>"`;

exports[`gux-table-beta #render should render table with vertical scroll 1`] = `"<gux-table-beta lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"gux-table gux-vertical-scroll\\"><div tabindex=\\"0\\" id=\\"gux-table-i\\" class=\\"gux-table-container\\"><table slot=\\"data\\"> <thead> <tr> <th>First name</th> <th>Last name</th> <th data-cell-numeric>Age</th> <th data-cell-action>Action</th> </tr> </thead> <tbody> <tr> <td>John</td> <td>Doe</td> <td data-cell-numeric>25</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>21</td> <td data-cell-action>Delete</td> </tr> <tr> <td>Jane</td> <td>Doe</td> <td data-cell-numeric>23</td> <td data-cell-action>Delete</td> </tr> </tbody> </table></div></div></gux-table-beta>"`;
